extensions/binding/org.eclipse.smarthome.binding.iota.test/src/test/java/org/eclipse/smarthome/binding/iota/IotaTestMocks.java

```diff
@@ -0,0 +1,238 @@
+                .withKind(ChannelKind.STATE).build();
+        iotaWalletThingHandler.updateThingMock(builder.withChannel(channel).build());
+
+        /**
+         * Add the corresponding ChannelConfig configuration for this channel
+         */
+        configWallet = channel.getConfiguration().as(ChannelConfig.class);
+        configWallet.setAddress(
+                "LJLGBRDIWNIFSOJXMTFVERCSSDEXTTXTHMGPIV9JY9R9WNSQUDHAIOBHRIJKOPFAXQCOTQGHGFRHASQW9TGBTDEGQB");
+        configWallet.setChannelUID(channel.getUID());
+        configWallet.setChannelStateUpdateListener(iotaWalletThingHandler);
+
+        /**
+         * Set the current value of the channel
+         */
+        configWallet.setValue(new NumberValue(isFloat));
+        iotaWalletThingHandler.addChannelDataByChannelUID(channel.getUID(), configWallet);
+
+        return configWallet;
+
+    }
+
+    protected ChannelConfig initializeTopicChannel(String value, String topic,
+            MockIotaTopicThingHandler iotaTopicThingHandler) {
+        /**
+         * Initialize thing and add channel
+         */
+        ThingBuilder builder = iotaTopicThingHandler.getThingBuilder();
+        Channel channel = ChannelBuilder
+                .create(new ChannelUID(iotaTopicThingHandler.getThing().getUID().toString() + ":groupid"),
+                        IotaBindingConstants.THING_TYPE_IOTA_TOPIC.toString())
+                .withKind(ChannelKind.STATE).build();
+        iotaTopicThingHandler.updateThingMock(builder.withChannel(channel).build());
+
+        /**
+         * Add the corresponding ChannelConfig configuration for this channel
+         */
+        configTopic = channel.getConfiguration().as(ChannelConfig.class);
+        configTopic.setChannelUID(channel.getUID());
+        configTopic.setChannelStateUpdateListener(iotaTopicThingHandler);
+
+        /**
+         * Set the current value of the channel
+         */
+        configTopic.setValue(new TextValue(value));
+        configTopic.setStateTopic(topic);
+        iotaTopicThingHandler.addChannelDataByChannelUID(channel.getUID(), configTopic);
+
+        return configTopic;
+
+    }
+
+    protected Bridge getBridge() {
+        return this.bridge;
+    }
+
+    protected Thing getThingWallet() {
+        return this.thingWallet;
+    }
+
+    protected Thing getThingTopic() {
+        return this.thingTopic;
+    }
+
+    protected ChannelConfig getConfigTopic() {
+        return configTopic;
+    }
+
+    protected ChannelConfig getConfigWallet() {
+        return configWallet;
+    }
+
+    protected IotaBridgeHandler getIotaBridgeHandler() {
+        return iotaBridgeHandler;
+    }
+
+    protected MockIotaWalletThingHandler getIotaWalletThingHandler() {
+        return iotaWalletThingHandler;
+    }
+
+    protected MockIotaTopicThingHandler getIotaTopicThingHandler() {
+        return iotaTopicThingHandler;
+    }
+
+    class MockIotaWalletThingHandler extends IotaWalletThingHandler {
```

Comment:
So far with what I tried, I haven't been able to do just that. Creating the thing with a pre-built channel will invoke the `startAutomaticRefresh()` method while initializing the thing since there are channels, and then I get a good load of thread related errors and whatnot.

Reply:
Lets sort out these errors. There are only a few bindings adding dynamic channels later. Here, from a modelling pov I think its not necessary. Can you debug whats happening? Please post some error logs.